singleRestSimulator/src/main/java/com/cisco/oss/foundation/tools/simulator/rest/container/SimulatorEntity.java

```diff
@@ -40,7 +41,7 @@
 	public SimulatorEntity(int port) {
 		this.port = port;
 		simulatorResponses = new ArrayList<SimulatorResponse>();
-		allRequests = new ArrayList<SimulatorRequest>();
+		allRequests = Collections.synchronizedList(new ArrayList<SimulatorRequest>());
```

Comment:
@binyaf I think adding synchronized is costly cpu wise. As this is a read mostly collection - wouldn't CopyOnWrite be a better solution?